csharp/src/Apache.Arrow.Adbc/AdbcDriver11.cs

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+
+namespace Apache.Arrow.Adbc
+{
+    /// <summary>
+    /// This provides a common interface for vendor-specific driver
+    /// initialization routines.
+    /// </summary>
+    public abstract class AdbcDriver11 : IDisposable
+    {
+        ~AdbcDriver11() => Dispose(false);
+
+        /// <summary>
+        /// Returns the version of the ADBC spec supported by this driver.
+        /// </summary>
+        public virtual int Version => AdbcVersion.Version_1_0_0;
```

Comment:
Should this property be available in all classes? Or just use .NET type management?
I see `AdbcDriver.DriverVersion` is the functional equivalent of this property. If we inherit from `AdbcDriver`, I guess we should stick to `DriverVersion`. Personally, I'd prefer this property be named `AdbcVersion` to disambiguate between driver, dbms, etc? Especially since it's an `int`.

Reply:
I think a consumer that cares would check this when loading the driver and remember the value. Putting it on all the types seems like it would be annoying for the implementer.